### ISSUE TITLE
Step into/step over/continue should not break first time at current IP if there's a breakpoint

### ISSUE THREAD

**Issue by user202729** [x64dbg/x64dbg#2324]
- Debugger version: Apr 12 2020.
- Operating system version and Service Pack: Windows 10, 64 bit
- Brief description of the issue: (title)
- Elaborate reproduction steps for the bug/issue being reported:

* Load any executable.
* Set a breakpoint at the current IP position.
* Press F7, F8 or F9.
* Observe that the IP doesn't move. it only moves the next time F7/8/9 is pressed.

The behavior is different in OllyDbg. Generally if the user press F7/8/9, they should be able to see the breakpoint there.

**Comment by mrexodia** [comment]
Yeah this is currently the intended behavior snd it’s very difficult to fix

**Comment by user202729** [comment]
Isn't it easy for F7? Just step over twice if there's a breakpoint at the current position? Or is there any problem with string or JMP command?

Also, (I don't understand the internal of x64dbg) why does it work the second time, but not the first time? The IP of the debugged process remains the same.

**Comment by flarejune** [comment]
Check here:
https://github.com/x64dbg/TitanEngine/blob/51ba022c29d3faae424ac6c51c0d7fb0a4cb5b84/TitanEngine/TitanEngine.Debugger.DebugLoop.cpp#L581


if(!HasBPX(DBGEvent.u.Exception.ExceptionRecord.ExceptionAddress)){
     engineStepCallBack
}

"HasBPX" Check for hardware breakpoints or software breakpoints at ExceptionAddress
